pallets/identity/src/auth.rs

```diff
@@ -187,7 +196,24 @@ impl<T: Config> Module<T> {
         }
 
         // Run custom per-type validation and updates.
-        accepter(auth.authorization_data, auth.authorized_by)?;
+        let res = accepter(auth.authorization_data, auth.authorized_by);
+
+        if res.is_err() {
+            let mut updated_auth = Self::ensure_authorization(target, auth_id)?;
+            // decrement
+            updated_auth.count -= 1;
+
+            // check if count is zero
+            if updated_auth.count == 0 {
+                <Authorizations<T>>::remove(&target, auth_id);
+                <AuthorizationsGiven<T>>::remove(updated_auth.authorized_by, auth_id);
+            }
+
+            // update authorization
+            <Authorizations<T>>::insert(&target, auth_id, updated_auth);
```

Comment:
When count is zero, the authorization will be remove and inserted again.  Only do the insert when `count > 0`.